test_util/cluster.py

```diff
@@ -184,6 +202,85 @@ def check_ssh(self, hosts):
         """
         self.ssher.remote_cmd(hosts, ['echo'])
 
+    def mesos_metrics_snapshot(self, host):
+        """Return a snapshot of the Mesos metrics for host."""
+        if host in self.masters:
+            port = 5050
+        else:
+            port = 5051
+
+        with self.ssher.tunnel(host) as tunnel:
+            return json.loads(
+                tunnel.remote_cmd(
+                    curl_cmd + ['{}:{}/metrics/snapshot'.format(host.private_ip, port)]
+                ).decode('utf-8')
+            )
+
+    def zk_mode(self, host):
+        """Return the mode of the ZooKeeper instance on host."""
+        with self.ssher.tunnel(host) as tunnel:
+            stat_out = tunnel.remote_cmd([
+                'echo', 'stat', '|', '/opt/mesosphere/bin/toybox', 'nc', 'localhost', '2181'
+            ])
+        for message in (l.strip().split(b':', 2) for l in stat_out.split(b'\n')):
+            if message[0] != b'Mode':
+                continue
+            mode = message[1].strip()
+            if mode not in (b'leader', b'follower', b'standalone'):
+                raise Exception('Unexpected ZooKeeper mode {} on host {}'.format(mode, host))
+            return mode.decode('utf-8')
+        raise Exception('ZooKeeper mode not found on host {}'.format(host))
+
+    def master_zk_modes(self):
+        """Return (master, zk_mode(master)) for each master."""
+        master_modes = [(master, self.zk_mode(master)) for master in self.masters]
```

Comment:
Comment not for fixing now

Actually discovered we have some similar code in test_endpoint (https://github.com/dcos/dcos/blob/master/packages/dcos-integration-test/extra/test_endpoints.py#L58) about checking for status, as well as in bootstrap for exhibitor waiting (https://github.com/dcos/dcos/blob/master/packages/bootstrap/extra/dcos_internal_utils/exhibitor.py#L67)

Eventually we should probably standardize / normalize those three down to one.